packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -66,7 +66,7 @@ class DaemonCommand extends FlutterCommand {
 
 typedef DispatchCommand = void Function(Map<String, dynamic> command);
 
-typedef CommandHandler = Future<dynamic> Function(Map<String, dynamic> args);
+typedef CommandHandler = FutureOr<dynamic> Function(Map<String, dynamic> args);
```

Comment:
FWIW, I would strongly prefer we never use `FutureOr`. It leads to all kinds of issues later.

Reply:
not quite sure i understand where this ends up mattering

Reply:
Removed, its not necessary